From e3f6b6175d17de692f55f53cb079c840b947d47d Mon Sep 17 00:00:00 2001
From: saranya <saranya.gopal@intel.com>
Date: Wed, 14 Aug 2019 14:33:14 +0530
Subject: [PATCH] EVS: Changes to enable evs in Celadon platform

This patch adds relevant changes to evs config,
app, manager and sampledriver to enable EVS feature
in Celadon with USB camera.

Test: Check if EVS preview comes when gear is park or reverse

Change-Id: I9d6a9d85db81533add204e864d39952bafa240a1
Tracked-On: OAM-85033
Signed-off-by: saranya <saranya.gopal@intel.com>
---
 evs/app/Android.mk                                           | 3 ++-
 evs/app/EvsStateControl.cpp                                  | 5 ++++-
 evs/app/config.json                                          | 2 +-
 evs/app/evs_app.rc                                           | 2 +-
 evs/manager/android.automotive.evs.manager@1.0.rc            | 2 +-
 .../android.hardware.automotive.evs@1.0-sample.rc            | 2 +-
 6 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/evs/app/Android.mk b/evs/app/Android.mk
index cecbba59..a9480c5a 100644
--- a/evs/app/Android.mk
+++ b/evs/app/Android.mk
@@ -20,6 +20,7 @@ LOCAL_SRC_FILES := \
     RenderPixelCopy.cpp
 
 LOCAL_SHARED_LIBRARIES := \
+    libbase \
     libbinder \
     libcutils \
     liblog \
@@ -80,4 +81,4 @@ LOCAL_REQUIRED_MODULES := \
     config.json \
     CarFromTop.png \
     LabeledChecker.png
-include $(BUILD_PHONY_PACKAGE)
\ No newline at end of file
+include $(BUILD_PHONY_PACKAGE)
diff --git a/evs/app/EvsStateControl.cpp b/evs/app/EvsStateControl.cpp
index 89222603..9022780c 100644
--- a/evs/app/EvsStateControl.cpp
+++ b/evs/app/EvsStateControl.cpp
@@ -26,6 +26,8 @@
 #include <utils/SystemClock.h>
 #include <binder/IServiceManager.h>
 
+#include <android-base/properties.h>
+
 static bool isSfReady() {
     const android::String16 serviceName("SurfaceFlinger");
     return android::defaultServiceManager()->checkService(serviceName) != nullptr;
@@ -245,13 +247,14 @@ bool EvsStateControl::selectStateForCurrentConditions() {
     // Choose our desired EVS state based on the current car state
     // TODO:  Update this logic, and consider user input when choosing if a view should be presented
     State desiredState = OFF;
+    bool allowPark = android::base::GetBoolProperty("persist.sys.evs.park", false);
     if (mGearValue.value.int32Values[0] == int32_t(VehicleGear::GEAR_REVERSE)) {
         desiredState = REVERSE;
     } else if (mTurnSignalValue.value.int32Values[0] == int32_t(VehicleTurnSignal::RIGHT)) {
         desiredState = RIGHT;
     } else if (mTurnSignalValue.value.int32Values[0] == int32_t(VehicleTurnSignal::LEFT)) {
         desiredState = LEFT;
-    } else if (mGearValue.value.int32Values[0] == int32_t(VehicleGear::GEAR_PARK)) {
+    } else if (mGearValue.value.int32Values[0] == int32_t(VehicleGear::GEAR_PARK) && allowPark) {
         desiredState = PARKING;
     }
 
diff --git a/evs/app/config.json b/evs/app/config.json
index 0d968f5b..ec25edfa 100644
--- a/evs/app/config.json
+++ b/evs/app/config.json
@@ -15,7 +15,7 @@
   },
   "cameras" : [
     {
-      "cameraId" : "/dev/video1",
+      "cameraId" : "/dev/video0",
       "function" : "reverse, park",
       "x" : 0.0,
       "y" : -40.0,
diff --git a/evs/app/evs_app.rc b/evs/app/evs_app.rc
index f155e11f..557d5266 100644
--- a/evs/app/evs_app.rc
+++ b/evs/app/evs_app.rc
@@ -3,4 +3,4 @@ service evs_app /system/bin/evs_app
     priority -20
     user automotive_evs
     group automotive_evs
-    disabled # will not automatically start with its class; must be explictly started.
+    #disabled # will not automatically start with its class; must be explictly started.
diff --git a/evs/manager/android.automotive.evs.manager@1.0.rc b/evs/manager/android.automotive.evs.manager@1.0.rc
index f1b58ede..89df133c 100644
--- a/evs/manager/android.automotive.evs.manager@1.0.rc
+++ b/evs/manager/android.automotive.evs.manager@1.0.rc
@@ -4,4 +4,4 @@ service evs_manager /system/bin/android.automotive.evs.manager@1.0
     user automotive_evs
     group automotive_evs
     onrestart restart evs_app
-    disabled # will not automatically start with its class; must be explictly started.
+    #disabled # will not automatically start with its class; must be explictly started.
diff --git a/evs/sampleDriver/android.hardware.automotive.evs@1.0-sample.rc b/evs/sampleDriver/android.hardware.automotive.evs@1.0-sample.rc
index 445a3a0c..71d6b021 100644
--- a/evs/sampleDriver/android.hardware.automotive.evs@1.0-sample.rc
+++ b/evs/sampleDriver/android.hardware.automotive.evs@1.0-sample.rc
@@ -4,4 +4,4 @@ service evs_driver /system/bin/android.hardware.automotive.evs@1.0-sample
     user graphics
     group automotive_evs camera
     onrestart restart evs_manager
-    disabled # will not automatically start with its class; must be explictly started.
+    #disabled # will not automatically start with its class; must be explictly started.
-- 
2.17.1

